### PR TITLE
Refactor import hints parser

### PR DESCRIPTION
**Pull request by Shatur**
In this PR, I reworked the parsing of import hints. Here is what I did:

* Material import hints only worked in the following order: `-alpha` `-vcol`. This is a bug. I replaced the parsing with a loop and the order is no longer important.
* Replaced `_fixstr` and `_teststr`, which contained duplicate code, with `_parse_hint` that changes passed name and returns the obtained hint.
* Enhanced the hints detection algorithm. Now it finds a hint and make sure that there are no letters around it. Supersedes #43563.
* Now the hints are enums and they are parsed only once, which speeds up the import. Also, all import hints are now in one place.

### CODE REVIEW

**Comment by seadra** [comment]
Does this fix #42041?

**Comment by Shatur95** [comment]
> Does this fix #42041?

Unfortunately no. But I'm interested in a fix. We will work on it as soon as this PR is merged!

**Comment by fire** [comment]
I like his. Will wait for the completion of the pr.

**Comment by Shatur95** [comment]
> I like this. Will wait for the completion of the pr.

The PR is ready, waiting for a review. I will send the fix in a separate PR to simplify review process.

**Review by fire — CHANGES_REQUESTED** [review]
This is better.

I only reviewed the design of the code.

Did you systematically check the test cases? Please reply back about that and then I can approve.

Minor things:

* Our coding style puts braces { } in the switch.

**Comment by Shatur95** [comment]
> Did you systematically check the test cases? Please reply back about that and then I can approve.

I tested it before and after changes the on submitted file in https://github.com/godotengine/godot-tests/pull/22 for material hints and on https://github.com/godotengine/godot-tests/blob/master/tests/blend_export/blend/32678-respect-import-hints-with-empty.blend for all other hints

> Our coding style puts braces { } in the switch.

Thanks, fixed!

**Review by fire — DISMISSED** [review]

**Comment by fire** [comment]
I'm not sure about the status of this, this pr was limbo'ed.

File: core/string/ustring.h
Comment (AThousandShips) [inline core/string/ustring.h:538]: This exists as `is_ascii_char` under `char_utils.h` now

**Pull request closed without merging**
